src/cogent3/app/sample.py

```diff
@@ -967,6 +976,22 @@ def main(self, aln: AlignedSeqsType) -> T:
         gaps_per_seq = aln.count_gaps_per_seq()
         length = len(aln)
         keep = [n for n, c in gaps_per_seq.items() if c / length < self._gap_fraction]
+
+        from cogent3.core import new_alignment
+
+        if self._ambig_fraction is not None:
+            if isinstance(aln, new_alignment.Alignment):
+                ambigs_per_seq = aln.count_ambiguous_per_seq()
+                keep = [
+                    n for n in keep if ambigs_per_seq[n] / length < self._ambig_fraction
+                ]
+            else:
```

Comment:
we don't do warnings within apps, better to just return a NotCompleted that way they can't help but notice!